capstone/src/main/java/com/step902020/capstone/Recommender.java

```diff
@@ -0,0 +1,73 @@
+
+package com.step902020.capstone;
+
+import com.step902020.capstone.security.CurrentUser;
+import java.io.IOException;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.util.Pair;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.function.Function;
+
+public class Recommender {
+
+  public static <E, U> List<E> recommend (U targetUser, List<U> users, Function<U, List<E>> getItemList, int numNeeded) {
```

Comment:
Definitely add Javadoc documenting what the generic types `E` and `U` are expected to be.